web/packages/teleterm/src/ui/ConnectMyComputer/CompatibilityPromise.tsx

```diff
@@ -0,0 +1,158 @@
+  const appMajorVersion = getMajorVersion(appVersion);
+  const connectMyComputerReleaseMajorVersion = getMajorVersion(
+    CONNECT_MY_COMPUTER_RELEASE
+  );
+
+  // offer a downgrade only to a release that has 'Connect My Computer'
+  const isAppDowngradePossible =
+    clusterMajorVersion >= connectMyComputerReleaseMajorVersion;
+  const downgradeAppTo =
+    clusterMajorVersion === connectMyComputerReleaseMajorVersion
+      ? CONNECT_MY_COMPUTER_RELEASE
+      : `${clusterMajorVersion}.x.x`;
+
+  let $content: JSX.Element;
+  if (appMajorVersion > clusterMajorVersion) {
+    $content = (
+      <>
+        , clusters don't support clients that are on a newer major version. If
+        you wish to connect your computer,{' '}
+        {isAppDowngradePossible && (
+          <>downgrade the app to {downgradeAppTo} or </>
+        )}
+        upgrade the cluster to version {appMajorVersion}.x.x.
+      </>
+    );
+  }
+  if (appMajorVersion < clusterMajorVersion) {
+    $content = (
+      <>
+        , clusters don't support clients that are more than one major version
+        behind. If you wish to connect your computer, upgrade the app to{' '}
+        {clusterMajorVersion}.x.x.
+      </>
+    );
+  }
+
+  return (
+    <>
+      <Alert>Detected an incompatible agent version.</Alert>
+      <Text>
+        The cluster is on version {serverVersion} while Teleport Connect is on
+        version {appVersion}. Per our{' '}
+        <Link
+          href="https://goteleport.com/docs/faq/#version-compatibility"
+          target="_blank"
+        >
+          compatibility promise
+        </Link>
+        {$content}
+      </Text>
+      <ButtonPrimary
+        mt={3}
+        mx="auto"
+        type="button"
+        as="a"
+        style={{ display: 'flex', width: 'fit-content' }}
+        href="https://goteleport.com/download"
+        target="_blank"
+      >
+        Visit the downloads page
+      </ButtonPrimary>
+    </>
+  );
+}
+
+export function UpgradeAgentSuggestion(): JSX.Element {
+  const ctx = useAppContext();
+  const workspaceContext = useWorkspaceContext();
+  const { isNonCompatibleAgent } = useConnectMyComputerContext();
+  const cluster = ctx.clustersService.findCluster(
+    workspaceContext.rootClusterUri
+  );
+  const { serverVersion } = cluster;
+  const { appVersion, isLocalBuild } =
+    ctx.mainProcessClient.getRuntimeSettings();
+
+  const isClusterAlreadyOnNewerVersion =
+    compareSemVers(appVersion, serverVersion) === 1;
+  if (isNonCompatibleAgent || isClusterAlreadyOnNewerVersion || isLocalBuild) {
+    return null;
+  }
+
+  return (
+    <Alert kind="info">
+      <Text>
+        The agent is running version {appVersion} of Teleport. Consider
+        upgrading it to {serverVersion} by updating Teleport Connect.{' '}
+        <Link href="https://goteleport.com/download" target="_blank">
+          Visit the downloads page.
+        </Link>
```

Comment:
IMHO we should either make only "the downloads page" the anchor (instead of also including "Visit" and the dot) or remove the last sentence completely and instead link "updating Teleport Connect" to the downloads page.

Reply:
I'm a fan of your first option as it makes the link a bit more explicit that you are visiting something and not updating it by clicking the link. But yes, don't include "Visit" in the link

Reply:
Thanks, I made only "the downloads page" the anchor.